examples/remote-settings-cli/src/downloader.rs

```diff
@@ -0,0 +1,289 @@
+        let capabilities: Value = self
+            .client
+            .get(REMOTE_SETTINGS_URL)
+            .send()
+            .await?
+            .json()
+            .await?;
+
+        let attachment_base_url = capabilities["capabilities"]["attachments"]["base_url"]
+            .as_str()
+            .context("No attachment base URL")?;
+
+        // Download each attachment
+        if let Some(records) = changes.as_array() {
+            for record in records {
+                if let Some(attachment) = record["attachment"].as_object() {
+                    if let (Some(location), Some(id)) =
+                        (attachment["location"].as_str(), record["id"].as_str())
+                    {
+                        let attachment_path = attachments_dir.join(id);
+                        let meta_path = attachments_dir.join(format!("{}.meta.json", id));
+
+                        // Save meta data
+                        std::fs::write(&meta_path, serde_json::to_string(&attachment)?)?;
+
+                        // Download attachment
+                        let bytes = self
+                            .client
+                            .get(&format!("{}{}", attachment_base_url, location))
+                            .send()
+                            .await?
+                            .bytes()
+                            .await?;
+
+                        std::fs::write(&attachment_path, bytes)?;
+                    }
+                }
+            }
+        }
+
+        Ok(())
+    }
+
+    pub(crate) async fn download_all(&self) -> Result<HashMap<String, CollectionData>> {
+        std::fs::create_dir_all(&self.output_dir)?;
+
+        let mut collections_data = HashMap::new();
+        let mut futures = FuturesUnordered::new();
+
+        let style = ProgressStyle::default_bar()
+            .template("[{elapsed_precise}] {bar:40.cyan/blue} {msg}")
+            .unwrap();
+
+        for collection in COLLECTIONS {
+            let pb = self.multi_progress.add(ProgressBar::new(100));
+            pb.set_style(style.clone());
+
+            futures.push(self.fetch_collection(collection.to_string(), pb));
+        }
+
+        while let Some(result) = futures.next().await {
+            match result {
+                Ok((collection, data)) => {
+                    collections_data.insert(collection, data);
+                }
+                Err(e) => {
+                    return Err(e);
+                }
+            }
+        }
+
+        Ok(collections_data)
+    }
+
+    pub(crate) async fn run(&self, dry_run: bool, create_pr: bool) -> Result<()> {
+        if dry_run && create_pr {
+            anyhow::bail!("Cannot use --dry-run with --create-pr");
+        }
+
+        let collections = self.download_all().await?;
+
+        let packaged_data = PackagedData {
+            collections: collections.clone(),
+        };
+
+        let output_file = self.output_dir.join("defaults.json");
```

Comment:
Why not having a folder with a file per collection? 
This file will end up being huge. Sometimes developers will want to open it just to double check its content compared to the server, etc.

Reply:
It's easier to go through one file and check if a collection exist. If we have multiple, I can check if the file exists, but the name could change etc. 

I am not against it, and also don't like the "one large file", but having multiple seem to have too many downsides?

Reply:
You seem convinced 😊 Maybe we can ask @Standard8 since his team seems interested to be able to modify the dump manually ([source](https://docs.google.com/document/d/1IIjopsfRLFlems-qDMDFO1HvSTeIOtPDHQolcuSTFvQ/edit#heading=h.t7qv62q3ujwt))

Note: The name cannot change, we cannot rename collections in Remote Settings.

Reply:
Desktop's `last_modified.json` file (generated with [gen_last_modified.py](https://searchfox.org/mozilla-central/source/services/settings/dumps/gen_last_modified.py)) has a list of the collections downloaded plus their timestamps. This allows you to look up in that file if the collection was downloaded, and then also check the timestamp to see if it is more recent than the local cache (e.g. for updating from local dumps when an update is released).

It is a little more work, but I think it would allow us to avoid having to load and parse the whole collection in one go, which might also have benefits especially for the invalid cache / update only what's needed on update cases.

From a developer perspective on the search team on desktop, we have a few scenarios that I think would lean towards separate files:

* We sometimes look back at the history of a collection to check roughly when it was updated. Although the date of the dump isn't necessarily the "live" date, if can give us a timeframe.
  * Similarly, if we're looking at changesets that have landed, it is easier to see that a file has changed, rather than having to dig through a large file diff for specific changes.
* For various collections, we will edit the collection locally and test it, before pushing changes to remote settings.
  * This allows us to test locally, and also run automated tests.
  * search-config-v2 and search-telemetry-v2 are quite complex for remote settings to display, hence we end up manually editing and uploading just the edited file.
* We quite often want to check or refer to something in the collection, this is easier to do in our local code editors. It is probably viable with a larger file, but they do tend to slow editors down / not be supported as well.